### PR TITLE
BUG: Ignore chartsheets

### LINKED ISSUE #41448: BUG: AttributeError raised when reading from excel file containing chart sheet
- [Y] I have checked that this issue has not already been reported.

- [Y] I have confirmed this bug exists on the latest version of pandas.

- [N] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example
```python
import pandas
pandas.read_excel('chartsheet.xlsx', sheet_name=None)
```

Example sheet file:
[chartsheet.xlsx](https://github.com/pandas-dev/pandas/files/6470217/chartsheet.xlsx)

#### Problem description
When reading an excel file that contains a chart sheet (a sheet that contains a chart and no table), the following exception was raised:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-89c95e3cb7b1> in <module>
      1 import pandas
----> 2 pandas.read_excel('chartsheet.xlsx', sheet_name=None)

D:\Softwares\Anaconda\lib\site-packages\pandas\util\_decorators.py in wrapper(*args, **kwargs)
    297                 )
    298                 warnings.warn(msg, FutureWarning, stacklevel=stacklevel)
--> 299             return func(*args, **kwargs)
    300 
    301         return wrapper

D:\Softwares\Anaconda\lib\site-packages\pandas\io\excel\_base.py in read_excel(io, sheet_name, header, names, index_col, usecols, squeeze, dtype, engine, converters, true_values, false_values, skiprows, nrows, na_values, keep_default_na, na_filter, verbose, parse_dates, date_parser, thousands, comment, skipfooter, convert_float, mangle_dupe_cols, storage_options)
    365             skipfooter=skipfooter,
    366             convert_float=convert_float,
--> 367             mangle_dupe_cols=mangle_dupe_cols,
    368         )
    369     finally:

D:\Softwares\Anaconda\lib\site-packages\pandas\io\excel\_base.py in parse(self, sheet_name, header, names, index_col, usecols, squeeze, converters, true_values, false_values, skiprows, nrows, na_values, parse_dates, date_parser, thousands, comment, skipfooter, convert_float, mangle_dupe_cols, **kwds)
   1188             convert_float=convert_float,
   1189             mangle_dupe_cols=mangle_dupe_cols,
-> 1190             **kwds,
   1191         )
   1192 

D:\Softwares\Anaconda\lib\site-packages\pandas\io\excel\_base.py in parse(self, sheet_name, header, names, index_col, usecols, squeeze, dtype, true_values, false_values, skiprows, nrows, na_values, verbose, parse_dates, date_parser, thousands, comment, skipfooter, convert_float, mangle_dupe_cols, **kwds)
    490                 sheet = self.get_sheet_by_index(asheetname)
    491 
--> 492             data = self.get_sheet_data(sheet, convert_float)
    493             usecols = maybe_convert_usecols(usecols)
    494 

D:\Softwares\Anaconda\lib\site-packages\pandas\io\excel\_openpyxl.py in get_sheet_data(self, sheet, convert_float)
    546         data: List[List[Scalar]] = []
    547         last_row_with_data = -1
--> 548         for row_number, row in enumerate(sheet.rows):
    549             converted_row = [self._convert_cell(cell, convert_float) for cell in row]
    550             if not all(cell == "" for cell in converted_row):

AttributeError: 'Chartsheet' object has no attribute 'rows'
```

#### Expected Output
I expect that either the chart sheet is ignored, returned as some appropriate object, or a warning is shown that the chart sheet cannot be read, instead of raising the exception.

In my case I'm reading from many zipped archives of excel files, so manually removing the charts from the files is not an option.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.7.4.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : zh_CN.UTF-8
LOCALE           : None.None

pandas           : 1.2.4
numpy            : 1.19.4
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 51.0.0
Cython           : 0.29.14
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.22.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : None
pyxlsb           : 1.0.8
s3fs             : None
scipy            : 1.5.4
sqlalchemy       : None
tables           : None
tabulate         : 0.8.7
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None
</details>

### PR DESCRIPTION
**Pull request by ahawryluk**
- [x] closes #41448
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Issue #41448 relates specifically to our openpyxl engine, and is resolved by this commit. However, the bug also exists in our pyxlsb engine and cannot be resolved until the upstream https://github.com/willtrnr/pyxlsb/issues/33 is addressed. I propose closing #41448 and leaving the pyxlsb case as an xfail for now. Let me know if there's a better way to track this.

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
Thanks @ahawryluk, looks good. Can you have a look at the CI, and see if you can leave it green please?

**Comment by ahawryluk** [comment]
@datapythonista I think #41665 tries to address the CI failure on the numpydev build

**Comment by ahawryluk** [comment]
@datapythonista CI is all green now. Thanks for taking a look at this.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Changes look good - I wonder if we should be changing the documentation for the `sheet_names` argument:

> Strings are used for sheet names. Integers are used in zero-indexed sheet positions. Lists of strings/integers are used to request multiple sheets. Specify None to get all sheets.

Get all worksheets instead?

Also, what happens if the index specified by sheetnames is a chart-sheet? What is a previous sheet is a chart sheet - is the chart sheet counted in the indexing?

**Comment by ahawryluk** [comment]
Hi @rhshadrach, thanks for the questions. I've attempted to address them in the docstring without getting too wordy. String and integer inputs are interpreted with respect to `ExcelFile.sheet_names`, which now excludes chart sheets, so the accepted inputs should be internally consistent and the default `sheet_name=0` won't crash if there's a chart sheet in the first position.

**Comment by rhshadrach** [comment]
Thanks @ahawryluk - makes sense. One last question - what is the error raised if the sheet name (as a string) is a chart sheet? If it's sensible, can you add a test for this.

**Comment by ahawryluk** [comment]
@rhshadrach Good idea. It raises the same ValueError you would get for an invalid sheet name, so I've added the tests for loading by str or by an int out of range.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
lgtm - can you resolve conflict and move whatsnew note to 1.4

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @ahawryluk!
